dhall/src/Dhall/Import.hs

```diff
@@ -744,8 +748,55 @@ toHeader _ = do
     empty
 
 getCacheFile
-    :: (Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
+    :: (MonadCatch m, Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
 getCacheFile cacheName hash = do
+    cacheDirectory <- getOrCreateCacheDirectory cacheName
+
+    let cacheFile = cacheDirectory </> ("1220" <> show hash)
+
+    return cacheFile
+
+warnAboutMissingCaches  :: (MonadCatch m, Alternative m, MonadIO m) => m ()
+warnAboutMissingCaches = do
+    warnings <- traverse warnAboutMissingCache ["dhall", "dhall-haskell"]
+
+    case catMaybes warnings of
+        warn:_ -> liftIO (System.IO.hPutStrLn System.IO.stderr warn)
+        []     -> return () 
+      
+
+warnAboutMissingCache :: (MonadCatch m, Alternative m, MonadIO m) => FilePath -> m (Maybe String)
+warnAboutMissingCache cacheName = do
```

Comment:
It *have to be* another way to do it instead raw pattern matching (or using `maybe`) but i didn't get it :worried: